Smarty/Smarty.class.php

```diff
@@ -1543,7 +1543,8 @@ function validateCompilePath( $compilePath ) {
         }
 
         $isValid = false;
-        if ( $fd = @fopen( $compilePath, 'wb') ) {
+        // Open file for writing but do not truncate.
+        if ( $fd = @fopen( $compilePath, 'c') ) {
             $isValid = true;
             @fclose( $fd );
             @unlink($compilePath);
```

Comment:
That does sound more robust (I have no idea why it works this way :) Oh and the other weird thing here is that it opens the file for writing. And for a short period of time this compiled template is now on disk, completely empty, if it didn't already exist. That's not safe for concurrent processes either. For reference this was added in https://issues.civicrm.org/jira/browse/CRM-5890